### PR TITLE
Make consent request cancelable

### LINKED ISSUE #565: Allow non-blocking transaction confirmation process
When we `Policy.Ask()` for transaction review and give consent, the confirmation channel is blocking, it shouldn't otherwise it's preventing the application to quit. That shouldn't.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #565

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -856,25 +841,6 @@ func (s *Service) signTx(token string, w http.ResponseWriter, r *http.Request, _
 
 	txHash, err := s.nodeForward.SendTx(r.Context(), tx, ty, cltIdx)
 	if err != nil {
```
Comment (wwestgarth) [inline service/service.go:843]: All the errors cases above `return` after calling `s.writeInternalError()` so I don't think we ever get to `s.policy.Report()` them?

**Comment by wwestgarth** [comment]
When I Ctrl+C when there is an transaction approval request, it just get:
```
Do you approve this transaction? (y/n) ^C
invalid answer "", expect "yes" or "no"
Do you approve this transaction? (y/n)
```
?

File: service/service.go (hunk 1 above)
Comment (ValentinTrinque) [reply service/service.go:843]: Good catch !

**Comment by ValentinTrinque** [comment]
> When I Ctrl+C when there is an transaction approval request, it just get:
> Do you approve this transaction? (y/n) ^C
> invalid answer "", expect "yes" or "no"
> Do you approve this transaction? (y/n)
> ?


@wwestgarth Yes that is right. It's a bit tricky to managed properly. I'll see what I can do.

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**